zerver/models.py

```diff
@@ -698,6 +698,41 @@ class RealmFilter(models.Model):
     class Meta:
         unique_together = ("realm", "pattern")
 
+    def clean(self) -> None:
```

Comment:
Can you update this commit message to explain to what extent this commit's adding of a `clean` method might reduce how much validation is done?  I don't remember enough details on how Django's model system `clean` methods work to be sure if this is a strict addition or not at this stage.

Reply:
This is a strict addition. We haven't removed any previous validations yet; they are removed in the next commit (`realm_filters: Remove regex based method for url and patern validation.`).

Django's full_clean that we call before saving a realm_filter in actions.py [does](https://kite.com/python/docs/django.db.models.Model.full_clean) the following:

> This method calls Model.clean_fields(), Model.clean(), and Model.validate_unique() (if validate_unique is True), in that order and raises a ValidationError that has a message_dict attribute containing errors from all three stages.